src/httpclient.c

```diff
@@ -479,6 +481,7 @@ static void usage(const char *progname)
             "  -m <method>  request method (default: GET). When method is CONNECT,\n"
             "               \"host:port\" should be specified in place of URL.\n"
             "  -o <path>    file to which the response body is written (default: stdout)\n"
+            "  -O <path>    file to which the response metadata is written (default: stderr)\n"
```

Comment:
I understand that this PR is WIP, but FWIW, I'm not sure if I agree with the direction of adding a flag to prevent UBSan errors from breaking CI.

If our intent is to address UB, we should fix them rather than adding a way to skip them. If our intent is instead to address address-related issues, then I do not think there's reason to turn on UBSan.

Reply:
Thank you for taking a look at this PR.

I was a bit confused by the type of errors. I thought there ware critical errors (which abort the process immediately) and non-critical errors (which does not abort the process)  and  thus were adding tricks to skip the non-critical ones, but in fact, ASan errors stop the process, but UBSan do not. So I think the timeline to introduce ASan and UBSan is like this:

1. Enable ASan (`-fsanitize=address`) and fixes the errors detected by it (this PR)
2. Enable UBSan (`-fsanitize=undefined`) and fixes the errors detected by it (another PR)


What do you think?